tests/order.spec.js

```diff
@@ -0,0 +1,373 @@
+                    .set('x-auth-token', token)
+                    .end((err, res) => {
+                        res.should.have.status(200);
+                        res.body.status.should.equal(200);
+                        res.body.data.should.be.an('array');
+                        done();
+                    });
+            });
+        });
+    });
+
+    describe('POST Purchase Order', () => {
+        let order;
+        beforeEach((done) => {
+
+            model.car = [{
+                id: 1,
+                owner: 1,
+                status: 'sold',
+                state: 'new',
+                created_on: Date.now(),
+                price: 200.5,
+                model: 'vs4-emmisteel',
+                body_type: 'car',
+                date_modified: Date.now(), 
+                manufacturer: 'toyota',  
+            }];
+
+            order = {
+                id: 1,
+                buyer: 2,
+                car_id: 1,
+                created_on: Date.now(),
+                amount: 200.05,
+                status: 'pending',
+                date_modified: Date.now(),
+            };
+
+            done();
+        });
+
+        afterEach((done) => {
+            model.car = [];
+            model.order = [];
+            done();
+        });
+
+        it('should post purchase order', (done) => {
+            chai.request(server)
+                .post('/api/v1/order')
+                .send(order)
+                .set('x-auth-token', token)
+                .end((err, res) => {
+                    res.should.have.status(201);
+                    res.body.status.should.equal(201);
+                    res.body.data.should.be.an('object');
+                    done();
+                });
+        });
+
+        it('should return 400 for purchase order', (done) => {
+            delete order.car_id;
+            chai.request(server)
+                .post('/api/v1/order')
+                .send(order)
+                .set('x-auth-token', token)
+                .end((err, res) => {
+                    res.should.have.status(400);
+                    res.body.status.should.equal(400);
+                    res.body.error.should.be.a('string');
+                    done();
+                });
+        });
+
+        it('should return 404 for purchase order', (done) => {
+            order.car_id = 2;
+            chai.request(server)
+                .post('/api/v1/order')
+                .send(order)
+                .set('x-auth-token', token)
+                .end((err, res) => {
+                    res.should.have.status(404);
+                    res.body.status.should.equal(404);
+                    res.body.error.should.be.a('string');
+                    done();
+                });
+        });
+    });
+
+    describe('Update Purchase Order', () => {
+        let amount;
+
+        beforeEach((done) => {
+
+            model.order = [{
+                id: 1,
```

Comment:
as said on the call, I will like you to have a model file which will house all instances of your model data.  instead of this. you should have the imported data here. i.e model.order = model.validorder

please ensure you implement this in all instance when data is needed in your test